Add documentation for the MySQL image

mnagy opened the pull request.

soltysh commented:
I'd propose moving `overview.adoc` and `mysql.adoc` to `others` subdir. And copy `openshift_sti_images` into this dir under name `sti`, so the final result should look like this:
```
- openshift_images/
  - sti/
    nodejs.adoc
    overview.adoc
    ruby.adoc
  - other/
    mysql.adoc
    overview.adoc
```

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+toc::[]
+
+== Overview
+Our OpenShift MySQL database image https://github.com/openshift/mysql[repository] contains the sources and Dockerfiles for building MySQL database images. The resulting images can be run by Docker.
```
s/Our/The/

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+Our OpenShift MySQL database image https://github.com/openshift/mysql[repository] contains the sources and Dockerfiles for building MySQL database images. The resulting images can be run by Docker.
+
+== Versions
+Current supported versions of MySQL:
```
Currently

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+
+[NOTE]
+====
+RHEL-7 image is not available as trusted build in https://index.docker.io[Docker Index]. In order to perform build or test action on a RHEL-7 based MySQL image, you need to run the build on properly subscribed RHEL-7 machine.
```
nothing is available as a trusted build in docker hub anymore, so just say that it's not available on docker hub.

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+====
+
+== Installation
+To build MySQL image, choose between CentOS7 or RHEL7 base image:
```
choose between *the* CentOS7 or RHEL7.....

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+=== CentOS-7 image
+
+----
+$ sti build https://github.com/ryanj/node-echo.git openshift/nodejs-010-centos7 nodejs-echo-app
```
this should be the ruby app, right?  and port 8080

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+----
+
+== Test
+OpenShift https://github.com/openshift/sti-nodejs/tree/master/0.10[STI-NodeJS] repository also provides STI test framework, which launches test to check functionality of a simple nodejs application built on top of sti-nodejs image.
```
looks like you copy+pasted this from the nodejs repo w/o fixing the names.

mnagy replied:
I used nodejs as a template, forgot to remove this whole section.

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,109 @@
+    will have full rights to the database.
+
+* *MYSQL_ROOT_PASSWORD* - Password for the root user. This variable is
+    optional. If not set, root user will not have any password set.
```
again we need to doc the exact behavior here, once you figure it out.

adellape commented:
@CowboysFan Please review the proposed `build_cfg` changes (also discussed in above comments).

CowboysFan commented:
@soltysh We're trying to avoid subdirectories, and I don't even think our build system will allow that. Both of these topic dirs and the topics within need quite a bit of restructuring in my opinion.

@adellape Let's go ahead and merge this and we'll take a look at it afterwards, taking suggestions from @soltysh into consideration. I'm working on the restructure so I'll just make it a part of that process. Thanks.

bparees commented:
@CowboysFan @adellape don't merge this yet, i made a bunch of comments that have not been addressed.

@mnagy ETA for updating this?

mnagy commented:
Sorry for the tardiness. PTAL.

I addressed the comments, but kept _build_cfg.yml as it was. I removed mentions of the test target, since I decided it probably should be a developer thing that's not in the openshift docs.

adellape commented:
Thanks, @mnagy. Merging.

FYI @tpoitras (per your related https://trello.com/c/jPzgJkic) and @CowboysFan (per `_build_cfg` changes for your restructure).

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,94 @@
+----
+
+=== CentOS-7 base image
+
```
or you can just pull the centos image from docker, no need to build it.

bparees commented on openshift_images/mysql.adoc:
```diff
@@ -0,0 +1,94 @@
+====
+
+== Usage
+To create a container running MySQL, you need to is run following commands.
```
s/is//

bparees commented:
I realize i'm post-merge here but i did make 2 minor comments, and we still need to be clearer about what happens if no admin password is set.

adellape commented:
@bparees That's cool. @tpoitras will likely be making a follow-up PR for this topic per his related docs card (https://trello.com/c/jPzgJkic) and should be able to address them.
